common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -319,4 +301,36 @@ public static void throwException(Throwable t) {
       DOUBLE_ARRAY_OFFSET = 0;
     }
   }
+
+  // This requires `majorVersion` and `_UNSAFE`.
+  static {
+    boolean _unaligned;
+    String arch = System.getProperty("os.arch", "");
+    if (arch.equals("ppc64le") || arch.equals("ppc64")) {
+      // Since java.nio.Bits.unaligned() doesn't return true on ppc (See JDK-8165231), but
+      // ppc64 and ppc64le support it
+      _unaligned = true;
+    } else {
+      try {
+        Class<?> bitsClass =
+          Class.forName("java.nio.Bits", false, ClassLoader.getSystemClassLoader());
+        if (_UNSAFE != null && 9 <= majorVersion && majorVersion <= 12) {
```

Comment:
Do you really want to cap this at Java 12? I presume Java 13 and beyond would behave similarly.

Reply:
Yes, it's because I only tested from Java 9 ~ 12. Since we don't know the future, the current intention is using the existing fallback logic to access `method` for Java 13+.

Reply:
I also open to remove that. Do you recommend to remove the cap?

Reply:
I'd remove it. Otherwise don't you fall back to the previous behavior in Java 13? more likely to work like Java 12 than 8.